src/map/lua/lua_baseentity.cpp

```diff
@@ -13386,7 +13386,14 @@ bool CLuaBaseEntity::hasClaim(CLuaBaseEntity* PTarget)
         return false;
     }
 
-    return battleutils::HasClaim(dynamic_cast<CBattleEntity*>(m_PBaseEntity), PBattleEntity);
+    auto* PTargetEntity = dynamic_cast<CBattleEntity*>(PTarget->GetBaseEntity());
+    if (!PTargetEntity)
+    {
+        ShowWarning("Attempting to check claim against invalid target type.");
+        return false;
+    }
+
+    return battleutils::HasClaim(PBattleEntity, PTargetEntity);
```

Comment:
Does this belong upstream? If not for any c++ changes that is PXI specific I would suggest either 1) a patch file system or 2) a standardized heading so when conflicts happen because of difference of files you know why this change is here